### ISSUE TITLE
Q: Is this actual training, or a knowledge base?

### ISSUE THREAD

**Issue by Lex-au** [mindverse/Second-Me#49]
Hi!

I read your readme, but haven't gone through the project files yet. Is this actual training, or just an end-to-end web app with knowledge base functionality (RAG)?

**Comment by yingapple** [comment]
Hey [Lex-au](https://github.com/Lex-au), we actually have a training process that allows you to fine-tune and own your own model. During inference, you can also combine it with RAG to provide more detailed and accurate information.

**Comment by Lex-au** [comment]
> Hey [Lex-au](https://github.com/Lex-au), we actually have a training process that allows you to fine-tune and own your own model. During inference, you can also combine it with RAG to provide more detailed and accurate information.

That’s super interesting, thanks for sharing. I noticed your repo mentions llama.cpp — as far as I know, it’s not technically possible to fine-tune a GGUF directly, since you’d typically need the original model weights in either PyTorch .bin or .safetensors format for that.

Could you clarify a bit on what you mean by “fine-tune and own your own model”? Are you referring to actual fine-tuning with gradient updates, or more along the lines of reward signal logging, preference-based reinforcement, or perhaps something like steerable generation using RAG?

Would love to understand more about how your setup handles that.

**Comment by akai-shuuichi** [comment]
> Hey [Lex-au](https://github.com/Lex-au), we actually have a training process that allows you to fine-tune and own your own model. During inference, you can also combine it with RAG to provide more detailed and accurate information.

Hi, I see that the current training capability is still marked as "Coming Soon". Does this mean that it is still in the form of RAG at present?

**Comment by yingapple** [comment]
> > Hey [Lex-au](https://github.com/Lex-au), we actually have a training process that allows you to fine-tune and own your own model. During inference, you can also combine it with RAG to provide more detailed and accurate information.
> 
> That’s super interesting, thanks for sharing. I noticed your repo mentions llama.cpp — as far as I know, it’s not technically possible to fine-tune a GGUF directly, since you’d typically need the original model weights in either PyTorch .bin or .safetensors format for that.
> 
> Could you clarify a bit on what you mean by “fine-tune and own your own model”? Are you referring to actual fine-tuning with gradient updates, or more along the lines of reward signal logging, preference-based reinforcement, or perhaps something like steerable generation using RAG?
> 
> Would love to understand more about how your setup handles that.

In one sentence: I am referring to actual fine-tuning involving gradient updates.

**Comment by yingapple** [comment]
> > Hey [Lex-au](https://github.com/Lex-au), we actually have a training process that allows you to fine-tune and own your own model. During inference, you can also combine it with RAG to provide more detailed and accurate information.
> 
> Hi, I see that the current training capability is still marked as "Coming Soon". Does this mean that it is still in the form of RAG at present?

No, we have very complicated pipeline. So we just released a part of fine-tuning pipeline. 
We are continuously organizing the code and gradually releasing complete versions that improve the effectiveness of fine-tuning.
